Cubical/Structures/AbGroup.agda

```diff
@@ -0,0 +1,47 @@
+{-# OPTIONS --cubical --safe #-}
+module Cubical.Structures.AbGroup where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.HLevels
+open import Cubical.Data.Sigma
+open import Cubical.Foundations.SIP renaming (SNS-PathP to SNS)
+
+open import Cubical.Structures.NAryOp
+open import Cubical.Structures.Group
+
+private
+  variable
+    ℓ : Level
+
+raw-abelian-group-structure : Type ℓ → Type ℓ
+raw-abelian-group-structure = raw-group-structure
+
+raw-abelian-group-is-SNS : SNS {ℓ} raw-abelian-group-structure _
+raw-abelian-group-is-SNS = raw-group-is-SNS
+
+abelian-group-axioms : (X : Type ℓ) → raw-abelian-group-structure X → Type ℓ
+abelian-group-axioms X _·_ = group-axioms X _·_ ×
+                             ((x y : X) → x · y ≡ y · x)
+
+abelian-group-structure : Type ℓ → Type ℓ
+abelian-group-structure = add-to-structure raw-abelian-group-structure abelian-group-axioms
+
+AbGroups : Type (ℓ-suc ℓ)
```

Comment:
I would call this `AbGroup` as each element will be an abelian group

Reply:
Done